BuildingMaterials/BuildingMaterials/Database.cs

```diff
@@ -0,0 +1,55 @@
+using System.Text.Json;
+using BuildingMaterials.Models;
+
+namespace BuildingMaterials;
+
+public class Database
+{
+    private const string DbFilePath = "db.json";
+
+    public List<MaterialModel> Materials { get; }
+
+    public Database()
+    {
+        if (!File.Exists(DbFilePath)) File.WriteAllText(DbFilePath, "[]");
+
+        var json = File.ReadAllText(DbFilePath);
+        Materials = JsonSerializer.Deserialize<List<MaterialModel>>(json) ?? [];
+    }
+
+    public int AddMaterial(MaterialModel material)
+    {
+        var maxId = Materials.Count != 0 ? Materials.Max(m => m.Id) : 0;
+
+        material.Id = maxId + 1;
+        Materials.Add(material);
+
+        SaveChanges();
+        return material.Id;
+    }
+
+    public bool UpdateMaterial(int id, MaterialModel material)
+    {
+        var index = Materials.FindIndex(m => m.Id == id);
```

Comment:
Why not FirstOrDefault?

Reply:
I can get new object of MaterialModel(Menu.cs line 152) and i want to assign a new object instead of the old one by index. For this and for other cases i dont need a full object